Sources/RedCat/ObservableStore.swift

```diff
@@ -7,31 +7,43 @@
 
 import Foundation
 
```

Comment:
Copying the entire app state can be expensive, in this instance even with copy-on-write optimized types. Better put an "oldValue" field into the state for those values where the old value is actually of interest. This can even be done with a property wrapper.

Also, if we share the state at all via the closure (rather than just notifying observers that they should review the store reference that they already hold), this should be done by reference, i.e. you should pass the store or some box type instead of just the state.